src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3290,3 +3326,126 @@ def _call_handler(self, node, parent):
         call.ast = node
 
         return call
+
+    def _subroutine_handler(self, node, parent):
+        '''Transforms an fparser2 Subroutine_Subprogram statement into a PSyIR
+        Routine node.
+
+        :param node: node in fparser2 parse tree.
+        :type node: :py:class:`fparser.two.Fortran2003.Subroutine_Subprogram`
+        :param parent: parent node of the PSyIR node being constructed.
+        :type parent: subclass of :py:class:`psyclone.psyir.nodes.Node`
+
+        :returns: PSyIR representation of node.
+        :rtype: :py:class:`psyclone.psyir.nodes.Routine`
+
+        '''
+        name = node.children[0].children[1].string
+        routine = Routine(name, parent=parent)
+
+        try:
+            sub_spec = first_type_match(node.content,
+                                        Fortran2003.Specification_Part)
+            decl_list = sub_spec.content
+            # TODO this if test can be removed once fparser/#211 is fixed
+            # such that routine arguments are always contained in a
+            # Dummy_Arg_List, even if there's only one of them.
+            if isinstance(node, Fortran2003.Subroutine_Subprogram) and \
+               isinstance(node.children[0].children[2],
+                          Fortran2003.Dummy_Arg_List):
+                arg_list = node.children[0].children[2].children
+            else:
+                # Routine has no arguments
+                arg_list = []
+        except ValueError:
+            # Subroutine has no Specification_Part so has no
+            # declarations. Continue with empty lists.
+            decl_list = []
+            arg_list = []
+        finally:
+            self.process_declarations(routine, decl_list, arg_list)
+
+        try:
+            sub_exec = first_type_match(node.content,
+                                        Fortran2003.Execution_Part)
+        except ValueError:
+            pass
```

Comment:
Add a comment saying that: Routines with no Execution part are still valid.

Reply:
Good idea.